Wikipedia/Code/LanguagesViewController.m

```diff
@@ -270,21 +269,38 @@ - (MWKLanguageLink*)languageAtIndexPath:(NSIndexPath*)indexPath {
 
 #pragma mark - UITableViewDelegate
 
-- (CGFloat)tableView:(UITableView*)tableView heightForFooterInSection:(NSInteger)section {
-    if ([self isPreferredSection:section] && self.languageFilter.filteredPreferredLanguages.count > 0) {
-        // collapse footer when empty, removing needless padding of "other" section from top of table
-        return WMFLanguagesSectionFooterHeight;
-    } else {
-        return 0.f;
+- (BOOL)shouldShowHeaderForSection:(NSInteger)section {
+    return ([self tableView:self.tableView numberOfRowsInSection:section] > 0);
+}
+
+- (UIView*)tableView:(UITableView*)tableView viewForHeaderInSection:(NSInteger)section {
+    if ([self shouldShowHeaderForSection:section]){
+        UIView* containerView             = [[UIView alloc] initWithFrame:CGRectZero];
```

Comment:
We should be using nibs and a class for header views.  Similar to how we use a cell subclass, instead of just creating and adding views to a stock UITableViewCell.

UITableView has APIs built for this:
https://developer.apple.com/library/ios/documentation/UIKit/Reference/UITableView_Class/index.html#//apple_ref/occ/instm/UITableView/registerNib:forHeaderFooterViewReuseIdentifier:
https://developer.apple.com/library/ios/documentation/UIKit/Reference/UITableView_Class/index.html#//apple_ref/occ/instm/UITableView/registerClass:forHeaderFooterViewReuseIdentifier:
https://developer.apple.com/library/ios/documentation/UIKit/Reference/UITableView_Class/index.html#//apple_ref/occ/instm/UITableView/dequeueReusableHeaderFooterViewWithIdentifier: